app/web/src/workers/webworker.ts

```diff
@@ -1928,38 +1935,42 @@ const niflheim = async (
     const finalAtoms = atomsForChangeSet(changeSetId);
     const atomsToUnlink: Array<Common> = [];
 
+    const processAtom = async (atom: AtomWithArrayBuffer) => {
+      let doc = getCachedDocument(atom.id, atom.checksum);
+      if (!doc) {
+        doc = decodeDocumentFromDB(atom.data);
+        setCachedDocument(atom.id, atom.checksum, doc);
+      }
+
+      postProcess(
+        workspaceId,
+        changeSetId,
+        atom.kind,
+        doc,
+        atom.id,
+        indexChecksum,
+        false,
+        false,
+        false,
+      );
+
+      bustCacheAndReferences(
+        workspaceId,
+        changeSetId,
+        atom.kind,
+        atom.id,
+        false,
+        true,
+      );
+    };
+
     for (const atom of finalAtoms) {
       // Atom is in the database, but not in the index? Delete it
       if (!atomSet.has(atomCacheKey(atom.id, atom.checksum))) {
         atomsToUnlink.push(atom);
       } else {
-        // Process the atom and bust cache and references related to it
-        let doc = getCachedDocument(atom.id, atom.checksum);
-        if (!doc) {
-          doc = decodeDocumentFromDB(atom.data);
-          setCachedDocument(atom.id, atom.checksum, doc);
-        }
-
-        postProcess(
-          workspaceId,
-          changeSetId,
-          atom.kind,
-          doc,
-          atom.id,
-          indexChecksum,
-          false,
-          false,
-          false,
-        );
-
-        bustCacheAndReferences(
-          workspaceId,
-          changeSetId,
-          atom.kind,
-          atom.id,
-          false,
-          true,
-        );
+        // Placing this in a promise to yield control back to the event loop
+        await processAtom(atom);
```

Comment:
I assume this is to let the `Promise.race` fail earlier, rather than have to wait? 🤔 Not sure I'm thinking about that correctly just yet

Reply:
This is to ensure messages to the webworker that has the lock can be processed in a timely way even while in the middle of a cold start. It's only tangentially related, mostly I found in testing that the leader election message would wait until a niflheim was finished before the message landed

Reply:
How does the `await` help them be done "timely" if they can just be done synchronously?

Reply:
The history of this one is that I was trying out a strategy that would try and round trip from the current tab to the tab with the lock through the shared worker, with a `ping`. But the pings would timeout if the tab with the lock was running a cold start, since it would never reply until the cold start was over (since the event loop was stuck processing the atoms). This made the pings respond. Although I removed that code in the end, it seems ideal to prevent the event loop from locking up, and this is simpler than adding a `globalThis.scheduler?.yield()` polyfill.